### PR TITLE
Update reqwest Dependency to Version 0.12.0 in google-cloud-storage Client (#251)

### PR DESCRIPTION
**Pull request by LenWilliamson**
**Description:**
This pull request addresses the issue #251, "Request for Migration to reqwest ^0.12.0 in google-cloud-storage Dependency". I have forked the repository, created a new branch, and made the necessary changes to update the reqwest dependency to version 0.12.0 within the google-cloud-storage client.

**Changes Made:**
- Updated reqwest dependency to version 0.12.4
- Confirmed compatibility with the google-cloud-storage client

**Additional Notes:**
I have ensured that the changes only affect the request dependencies within the storage client and have not made any unrelated modifications to the other crates. While I couldn't perform local testing please test these changes in the project environment.

**Request for Review:**
I kindly request the maintainer to review these changes and consider merging them into the main repository. Thank you for your attention to this matter. I am open to any feedback or suggestions for improvement.

Best regards,
Len

### CODE REVIEW

File: foundation/auth/Cargo.toml [hunk 1]
```diff
@@ -11,7 +11,7 @@ description = "Google Cloud Platform server application authentication library."
 
 [dependencies]
 tracing = "0.1"
-reqwest = { version = "0.11", features = ["json"], default-features = false }
+reqwest = { version = "0.12.4", features = ["json"], default-features = false }
```
Comment (yoshidan) [inline foundation/auth/Cargo.toml:14]: `charset` feature is required to use `text_with_charset`  at  0.12.4 
https://docs.rs/reqwest/0.12.4/reqwest/struct.Response.html#method.text_with_charset

Comment (LenWilliamson) [reply foundation/auth/Cargo.toml]: I added this feature to `foundation/auth/Cargo.toml`.

**Comment by yoshidan** [comment]
@LenWilliamson 

1. Could you upgrade `retry-policies` crate to fix interface error.
https://github.com/yoshidan/google-cloud-rust/actions/runs/8889195974/job/24407198542?pr=259#step:6:368

2. Could you upgrade  all the crate that uses `reqwest 0.11` to fix version mismatch error.
https://github.com/yoshidan/google-cloud-rust/actions/runs/8867531149/job/24346192853?pr=254

**Comment by yoshidan** [comment]
Thanks!

**Pull request merged**